ui/src/notebooks/shared/Resizer.tsx

```diff
@@ -0,0 +1,175 @@
+  })
+
+  const updateResultsStyle = (): void => {
+    if (bodyRef.current && resizingEnabled && visibility === 'visible') {
+      bodyRef.current.setAttribute('style', `height: ${size}px`)
+    } else {
+      bodyRef.current.setAttribute('style', '')
+    }
+  }
+
+  const handleUpdateVisibility = (panelVisibility: Visibility): void => {
+    onUpdate({panelVisibility})
+  }
+
+  const handleUpdateHeight = (panelHeight: number): void => {
+    onUpdate({panelHeight})
+  }
+
+  // Ensure results renders with proper height on initial render
+  useEffect(() => {
+    updateResultsStyle()
+  }, [])
+
+  // Update results height when associated props change
+  useEffect(() => {
+    updateResultsStyle()
+  }, [size, visibility, resizingEnabled])
+
+  // Update local height when context height changes
+  // so long as it is a different value
+  useEffect(() => {
+    if (height !== size) {
+      updateSize(height)
+    }
+  }, [height])
+
+  // Handle changes in drag state
+  useEffect(() => {
+    if (isDragging === true) {
+      dragHandleRef.current &&
+        dragHandleRef.current.classList.add(
+          'panel-resizer--drag-handle__dragging'
+        )
+    }
+
+    if (isDragging === false) {
+      dragHandleRef.current &&
+        dragHandleRef.current.classList.remove(
+          'panel-resizer--drag-handle__dragging'
+        )
+      handleUpdateHeight(size)
+    }
+  }, [isDragging])
+
+  const handleMouseMove = (e: MouseEvent): void => {
+    if (!bodyRef.current) {
+      return
+    }
+
+    const {pageY} = e
+    const {top} = bodyRef.current.getBoundingClientRect()
+
+    const updatedHeight = Math.round(Math.max(pageY - top, minimumHeight))
+
+    updateSize(updatedHeight)
+  }
+
+  const handleMouseDown = (): void => {
+    updateDragging(true)
+    const body = document.getElementsByTagName('body')[0]
+    body && body.classList.add('panel-resizer-dragging')
+
+    window.addEventListener('mousemove', handleMouseMove)
+    window.addEventListener('mouseup', handleMouseUp)
+  }
+
+  const handleMouseUp = (): void => {
+    updateDragging(false)
+    const body = document.getElementsByTagName('body')[0]
+    body && body.classList.remove('panel-resizer-dragging')
+
+    window.removeEventListener('mousemove', handleMouseMove)
+    window.removeEventListener('mouseup', handleMouseUp)
+  }
+
+  let body = children
+
+  if (!resizingEnabled) {
+    body = <div className="panel-resizer--empty">{emptyText}</div>
+  }
+
+  if (resizingEnabled && visibility === 'hidden') {
+    body = <div className="panel-resizer--empty">{hiddenText}</div>
+  }
+
+  return (
+    <div className="panel-resizer">
+      <ResizerHeader
+        emptyIcon={emptyIcon}
+        visibility={visibility}
+        onStartDrag={handleMouseDown}
+        dragHandleRef={dragHandleRef}
+        resizingEnabled={resizingEnabled}
+        additionalControls={additionalControls}
+        onUpdateVisibility={handleUpdateVisibility}
```

Comment:
this shouldn't be passed down

Reply:
note: this is going to be addressed in a clean up pass when adding the "CurrentPipe" context